test/new-e2e/agent-platform/install-script/install_script_test.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+package installscript
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/agent-platform/common"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	e2eOs "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+
+	"testing"
+)
+
+var osVersion = flag.String("osversion", "", "os version to test")
+var platform = flag.String("platform", "", "platform to test")
+var cwsSupportedOsVersion = flag.String("cws-supported-osversion", "", "list of os where CWS is supported")
+
+type installScriptSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+	cwsSupported bool
+}
+
+func TestInstallScript(t *testing.T) {
+	platformJSON := map[string]map[string]string{}
+	platformFileContent, err := os.ReadFile("../platforms.json")
+	if err != nil {
+		panic(fmt.Sprintf("failed to read platform file: %v", err))
+	}
+	err = json.Unmarshal(platformFileContent, &platformJSON)
+	if err != nil {
+		panic(fmt.Sprintf("failed to umarshall platform file: %v", err))
+	}
+	osVersions := strings.Split(*osVersion, ",")
+	cwsSupportedOsVersionList := strings.Split(*cwsSupportedOsVersion, ",")
+
+	for _, osVers := range osVersions {
+		osVers := osVers
+		cwsSupported := false
+		for _, cwsSupportedOs := range cwsSupportedOsVersionList {
+			if cwsSupportedOs == osVers {
+				cwsSupported = true
+			}
+		}
+
+		t.Run(fmt.Sprintf("test install script on %s", osVers), func(tt *testing.T) {
+			tt.Parallel()
+			fmt.Printf("Testing %s", osVers)
+			e2e.Run(tt, &installScriptSuite{cwsSupported: cwsSupported}, e2e.AgentStackDef(e2e.WithVMParams(ec2params.WithImageName(platformJSON[*platform][osVers], e2eOs.AMD64Arch, ec2os.DebianOS)), e2e.WithAgentParams(agentparams.WithAgentConfig("site: datadoghq.eu"))), params.WithStackName(fmt.Sprintf("install-script-test-%v-%v", os.Getenv("CI_PIPELINE_ID"), osVers)))
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
I would rather use the `VMStackDef` and explicitly call the install script. We do want to test the install script, and by using the `AgentStackDef` we delegate the install script call to the underlying code in `test-infra-definitions`. We might break the test by updating the implementation, that is hidden here.

Reply:
I hesitated a bit between the two. Since the agent used the install script to install the agent, to me it was not a problem to use it like that, if the installation could change it would indeed be a problem.
Using VMStackDef would imply rewriting some logic we wrote on `test-infra-definitions` (getting the correct install script command depending on the OS, setting the correct arguments to install the version from the current pipeline, ...). It would also need to find a way to expose the `Agent` [API](https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/pkg/utils/e2e/client/agent_cmd_runner.go) in the VMStackDef stack